URL.navigate fails with path in base URL

The following and several permutations fail attempting to concatenate a `tuple` with a `list`:

```
URL('http://localhost').navigate('a').navigate('b')
```
Result:

```
...
            else:  # relative path
>               new_path_parts = list(self.path_parts[:-1]) + dest.path_parts
E               TypeError: can only concatenate list (not "tuple") to list

boltons/urlutils.py:657: TypeError
```

I am a little unclear about exactly what *should* happen when 'a' has a trailing slash or not, but enough cases fail that I know *something* is wrong.
I will follow shortly with a PR with a number of test cases, some of which I know are wrong, so I am hoping I can get some clarification about what the correct behavior *should* be.

This is not strictly an issue with chaining the `navigate` calls, but rather ambiguity of what should happen with the base URL lacks a trailing '/' and failures that result from that.

Env data:
platform darwin
Python 3.6.5 (built w/pyenv)
pytest-3.5.1

--- a/boltons/urlutils.py
+++ b/boltons/urlutils.py
@@ -654,7 +654,8 @@ class URL(object):
             if dest.path.startswith(u'/'):   # absolute path
                 new_path_parts = list(dest.path_parts)
             else:  # relative path
-                new_path_parts = self.path_parts[:-1] + dest.path_parts
+                new_path_parts = list(self.path_parts[:-1]) \
+                               + list(dest.path_parts)
         else:
             new_path_parts = list(self.path_parts)
             if not query_params:

--- a/tests/test_urlutils.py
+++ b/tests/test_urlutils.py
@@ -271,11 +271,11 @@ def test_rel_navigate():
 @pytest.mark.parametrize(
     ('expected',            'base',             'a',    'b'), [
     ('https://host/b',      'https://host',     'a',    '/b'),
-    ('https://host/a/b',    'https://host',     'a',    'b'),
+    ('https://host/b',    'https://host',     'a',    'b'),
     ('https://host/a/b',    'https://host',     'a/',   'b'),
-    ('https://host/a/b',    'https://host',     '/a',   'b'),
+    ('https://host/b',    'https://host',     '/a',   'b'),
     ('https://host/a/b',    'https://host/a/',  None,   'b'),
-    ('https://host/a/b',    'https://host/a',   None,   'b'),
+    ('https://host/b',    'https://host/a',   None,   'b'),
 ])
 def test_chained_navigate(expected, base, a, b):
     """Chained :meth:`navigate` calls produces correct results."""

--- a/tests/test_urlutils.py
+++ b/tests/test_urlutils.py
@@ -268,25 +268,6 @@ def test_rel_navigate():
     return
 
 
-@pytest.mark.parametrize(
-    ('expected', 'base', 'paths'), [
-    ('https://host/b', 'https://host', ['a', '/b']),
-    ('https://host/b', 'https://host', ['a', 'b']),
-    ('https://host/a/b', 'https://host', ['a/', 'b']),
-    ('https://host/b', 'https://host', ['/a', 'b']),
-    ('https://host/a/b', 'https://host/a/', [None, 'b']),
-    ('https://host/b', 'https://host/a', [None, 'b']),
-])
-def test_chained_navigate(expected, base, paths):
-    """Chained :meth:`navigate` calls produces correct results."""
-    url = URL(base)
-
-    for path in paths:
-        url = url.navigate(path)
-
-    assert expected == url.to_text()
-
-
 def test_navigate():
     orig_text = u'http://a.b/c/d?e#f'
     orig = URL(orig_text)
@@ -330,6 +311,25 @@ def test_navigate():
     assert navd.to_text() == _dest_text
     navd = orig.navigate(_dest_text)
     assert navd.to_text() == _dest_text
+
+
+@pytest.mark.parametrize(
+    ('expected', 'base', 'paths'), [
+    ('https://host/b', 'https://host', ('a', '/b', )),
+    ('https://host/b', 'https://host', ('a', 'b', )),
+    ('https://host/a/b', 'https://host', ('a/', 'b', )),
+    ('https://host/b', 'https://host', ('/a', 'b', )),
+    ('https://host/a/b', 'https://host/a/', (None, 'b', )),
+    ('https://host/b', 'https://host/a', (None, 'b', )),
+])
+def test_chained_navigate(expected, base, paths):
+    """Chained :meth:`navigate` calls produces correct results."""
+    url = URL(base)
+
+    for path in paths:
+        url = url.navigate(path)
+
+    assert expected == url.to_text()
 
 
 # TODO: RFC3986 6.2.3 (not just for query add, either)

--- a/tests/test_urlutils.py
+++ b/tests/test_urlutils.py
@@ -268,6 +268,23 @@ def test_rel_navigate():
     return
 
 
+@pytest.mark.parametrize(
+    ('expected',            'base',             'a',    'b'), [
+    ('https://host/b',      'https://host',     'a',    '/b'),
+    ('https://host/a/b',    'https://host',     'a',    'b'),
+    ('https://host/a/b',    'https://host',     'a/',   'b'),
+    ('https://host/a/b',    'https://host',     '/a',   'b'),
+    ('https://host/a/b',    'https://host/a/',  None,   'b'),
+    ('https://host/a/b',    'https://host/a',   None,   'b'),
+])
+def test_chained_navigate(expected, base, a, b):
+    """Chained :meth:`navigate` calls produces correct results."""
+    if a is not None:
+        assert expected == URL(base).navigate(a).navigate(b).to_text()
+    else:
+        assert expected == URL(base).navigate(b).to_text()
+
+
 def test_navigate():
     orig_text = u'http://a.b/c/d?e#f'
     orig = URL(orig_text)

--- a/tests/test_urlutils.py
+++ b/tests/test_urlutils.py
@@ -269,20 +269,22 @@ def test_rel_navigate():
 
 
 @pytest.mark.parametrize(
-    ('expected',            'base',             'a',    'b'), [
-    ('https://host/b',      'https://host',     'a',    '/b'),
-    ('https://host/b',    'https://host',     'a',    'b'),
-    ('https://host/a/b',    'https://host',     'a/',   'b'),
-    ('https://host/b',    'https://host',     '/a',   'b'),
-    ('https://host/a/b',    'https://host/a/',  None,   'b'),
-    ('https://host/b',    'https://host/a',   None,   'b'),
+    ('expected', 'base', 'paths'), [
+    ('https://host/b', 'https://host', ['a', '/b']),
+    ('https://host/b', 'https://host', ['a', 'b']),
+    ('https://host/a/b', 'https://host', ['a/', 'b']),
+    ('https://host/b', 'https://host', ['/a', 'b']),
+    ('https://host/a/b', 'https://host/a/', [None, 'b']),
+    ('https://host/b', 'https://host/a', [None, 'b']),
 ])
-def test_chained_navigate(expected, base, a, b):
+def test_chained_navigate(expected, base, paths):
     """Chained :meth:`navigate` calls produces correct results."""
-    if a is not None:
-        assert expected == URL(base).navigate(a).navigate(b).to_text()
-    else:
-        assert expected == URL(base).navigate(b).to_text()
+    url = URL(base)
+
+    for path in paths:
+        url = url.navigate(path)
+
+    assert expected == url.to_text()
 
 
 def test_navigate():

